### PR TITLE
feat: Use `~/.osmosisd` in LocalOsmosis 

### PR DESCRIPTION
**Pull request by niccoloraspa**
## What is the purpose of the change

This PR:

- promotes the use `~/.osmosisd` folder for LocalOsmosis instead of an in-repo folder 

- modifies the LocalOsmosis flow:

```
make localnet-init 
# builds the docker image
# prompts the user to clean the ~/.osmosisd folder if needed

make localnet-start 
# or `make localnet-startd`

make localnet-stop

make localnet-clean
# I have replaced localnet-remove with this
```

- Renames ` localnet-remove` to `localnet-clean`

- Renames Makefile commands from `make localnet-COMMAND-state-export` to `make localnet-state-export-COMMAND` (a little bit easier to edit, but we can revert it in case of problems). 
We may have to update the `osmosis-installer` (cc: @czarcas7ic)

- Update docs

- **Bug fix** In the previous PR I haven't pushed the commit that renamed the `args.verbose` to `args.quiet` in the `testnetify.py` script. I have added it here. 

## Brief Changelog

- promotes the use `~/.osmosisd` folder for LocalOsmosis instead of an in-repo folder 

- Bug fix: remove references to `args.verbose` in `testnetify.py`

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?no
  - How is the feature or change documented? Updated docs in `tests/localosmosis/README.md`

### CODE REVIEW

**Comment by niccoloraspa** [comment]
cc: @p0mvn 

In case you need to update the workshop repo

File: tests/localosmosis/mainnet_state/stateprune.py [hunk 1]
```diff
@@ -1,34 +0,0 @@
-import json
```
Comment (niccoloraspa) [inline tests/localosmosis/mainnet_state/stateprune.py:1]: Is this file okay to be removed?
@czarcas7ic

Comment (czarcas7ic) [reply tests/localosmosis/mainnet_state/stateprune.py:1]: Yes!

**Review by czarcas7ic — APPROVED** [review]
This LGTM, thanks for implementing this!

**Pull request merged**